awx/main/tests/functional/api/test_inventory.py

```diff
@@ -60,6 +60,24 @@ def test_inventory_source_unique_together_with_inv(inventory_factory):
     is2.validate_unique()
 
 
+@pytest.mark.django_db
+def test_inventory_host_name_unique(scm_inventory, post, admin_user):
+    inv_src = scm_inventory.inventory_sources.first()
+    group = inv_src.groups.create(name='barfoo', inventory=scm_inventory)
+    group.save()
+    host1 = inv_src.hosts.create(name='barfoo', inventory=scm_inventory)
+    post(reverse('api:inventory_hosts_list', kwargs={'pk': host1.id}), admin_user, expect=400)
```

Comment:
You're getting a 400 error here because this `POST` request isn't actually creating the host (because you already created it on the line right above).  Instead, it's making a request like this (which is invalid):

```http
HTTP POST /api/v2/hosts/N/
{}
```

What you *want* is to make _this_ request:

```http
HTTP POST /api/v2/hosts/
{
    "name": "barfoo",
    "inventory_id": 1
}
```

You want to pass the payload as parameters for the request, and get back a 400.  Maybe something like this:

```python
group.save()
resp = post(
    reverse('api:inventory_hosts_list'),
    {
        'name': 'barfoo',
        'inventory_id': scm_inventory.id,
    }
    admin_user,
    expect=400
)
```

Additionally, "HTTP 400" can mean a lot of things.  To make sure you're actually hitting the code path you expect, you might want to just add an additional assertions here:

```python
assert "A Group with that name already exists." in resp.data.get('name', [])
```